lms/djangoapps/courseware/module_render.py

```diff
@@ -917,7 +918,8 @@ def get_module_by_usage_id(request, course_id, usage_id, disable_staff_debug_inf
     field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
         course_id,
         user,
-        descriptor
+        descriptor,
+        read_only=CrawlersConfig.is_crawler(request),
```

Comment:
The reason why I only did these two calls were because they were the relevant ones for the courseware index page rendering, which was the only page we're seeing impacted by crawler traffic at the moment. Not that it couldn't be done, but just that I wanted to confine this PR to an easily testable piece that fixed the observed problem.

I realize we can grab the request object deep in the stack, but I prefer to avoid that when I can, since I feel like it breaks an abstraction barrier and is a bit spooky. I prefer FieldDataCache just be made explicitly read_only or not at instantiation.